scripts/quests/otherAreas/A_Generous_General.lua

```diff
@@ -0,0 +1,290 @@
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET + 2,
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET + 3,
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET + 4,
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET + 5,
+    oldtonID.mob.GENEROUS_GENERAL_OFFSET + 6,
+}
+
+quest.reward =
+{
+    item  = xi.item.CHOPLIXS_COIF,
+    title = xi.title.MOBLIN_KINSMAN,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+                player:getQuestStatus(xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.AN_AFFABLE_ADAMANTKING) ~= xi.questStatus.QUEST_ACCEPTED and
+                player:getQuestStatus(xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.AN_UNDERSTANDING_OVERLORD) ~= xi.questStatus.QUEST_ACCEPTED and
+                player:getQuestStatus(xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.A_MORAL_MANIFEST) ~= xi.questStatus.QUEST_ACCEPTED and
+                player:getVar('BstHeadGearQuest_Conquest') < NextConquestTally() and
+                not player:hasItem(xi.item.CHOPLIXS_COIF) and
+                player:getMainLvl() >= 60
+        end,
+
+        [xi.zone.OLDTON_MOVALPOLOS] =
+            {
+            onZoneIn =
+            {
+                function(player, prevZone)
+                    local pos = player:getPos()
+                    if
+                        pos.x >= -323.0 and
+                        pos.y >= 6.0 and
+                        pos.z >= -261.0 and
+                        pos.x <= -321.0 and
+                        pos.y <= 9.0 and
+                        pos.z <= -258.0
+                    then
+                        return 60
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [60] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:begin(player) -- begin quest
+                    elseif option == 1 then
+                        player:setVar('BstHeadGearQuest_Conquest', NextConquestTally()) -- if player declines they must wait until next conquest tally
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Faulpie'] =
+            {
+                onTrigger = function(player, npc)
+                    local prog = quest:getVar(player, 'Prog')
+                    local partsWait = quest:getVar(player, 'partsWait')
+                    if prog == 0 then
+                        return quest:progressEvent(770) -- tells player what items are needed
+                    elseif prog == 1 then
+                        return quest:event(771):importantOnce() -- reminder to get buffalo hide and sheep leather or cancel quest
+                    elseif
+                        prog >= 2 and
```

Comment:
Loose condition here, what happens if partsWait == 0 and prog == 2?

Reply:
Hmm,

`partsWait` is set to `VanaMidnight()` at the same time (right after) `prog` is set to `2` with event 772.

`partsWait` is only set to `0` in event `775`, if player `prog` is `2`, then it is set to `3`. So we don't hit partsWait == 0 and prog == 2.

Are you seeing a possibility of partsWait not being set to `VanaMidnight` or being reset to `0` that I might be missing?